### PR TITLE
[build] Enable vendoring 

### PR DESCRIPTION
**Pull request by mansikulkarni96**
This PR ensures we enable vendoring in this repository, following the vendoring standard of the OpenShift org. 
go command now defaults to -mod=vendor for operations like go build that accept that flag, hence we do not need to set them explicitly.

### CODE REVIEW

File: hack/verify-vendor.sh [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+popd > /dev/null 2>&1
+
+ret=0
+
+pushd "${WMCB_ROOT}" > /dev/null 2>&1
+if ! _out="$(diff -Naupr -x "BUILD" -x "AUTHORS*" -x "CONTRIBUTORS*" vendor "${_wmcbtmp}/vendor")"; then
+   echo "Your Vendored results are different:" >&2
+   echo "${_out}" >&2
+   echo "Vendor verify failed." >&2
+   echo "${_out}" > vendordiff.patch
```
[the remark this answers is not in the document]
Comment (mansikulkarni96) [reply hack/verify-vendor.sh:37]: yes I have run it locally. It is not downloading them, the wmcb job failed due to CI must-gather issue, but I can see in the wni tests that it is not downloading the go mod deps -> [wni test log](https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_windows-machine-config-bootstrapper/230/pull-ci-openshift-windows-machine-config-bootstrapper-master-e2e-wni/1298983381840695296/artifacts/e2e-wni/e2e-wni/container-logs/test.log)

[the remark this answers is not in the document]
Comment (mansikulkarni96) [reply hack/verify-vendor.sh:37]: added vendordiff.patch to .gitignore

**Comment by VaishnaviHire** [comment]
LGTM

**Pull request merged**